src/components/FlowListItem.vue

```diff
@@ -0,0 +1,198 @@
+          size="xs"
+          class="flow-list-item__content-toggle"
+          :class="computedClasses.toggle"
+          inset
+          icon="ChevronDownIcon"
+        />
+
+        <span v-if="deploymentsCount > 0" class="flow-list-item__content-text">
+          {{ deploymentsCount }}  {{ toPluralString(localization.info.deployment, deploymentsCount) }}
+        </span>
+
+        <span v-else class="flow-list-item__content-text-none">
+          {{ localization.info.noDeployments }}
+
+          <ExtraInfoModal :title="localization.info.noDeployments">
+            <p-markdown-renderer :text="localization.info.deploymentsEmptyStateDescription(flow.name)" />
+            <template #actions>
+              <DocumentationButton :to="localization.docs.deployments" />
+            </template>
+          </ExtraInfoModal>
+        </span>
+      </div>
+    </StateListItem>
+
+    <keep-alive v-if="canExpand">
+      <p-auto-height-transition>
+        <template v-if="expanded">
+          <slot>
+            <DeploymentList :disabled="disabled" :filter="filter" class="flow-list-item__deployments" />
+          </slot>
+        </template>
+      </p-auto-height-transition>
+    </keep-alive>
+  </div>
+</template>
+
+<script lang="ts">
+  export default {
+    name: 'FlowListItem',
+    expose: [],
+    inheritAttrs: false,
+  }
+</script>
+
+<script lang="ts" setup>
+  import { toPluralString, useAttrsStylesAndClasses } from '@prefecthq/prefect-design'
+  import { useLocalStorage, useSubscriptionWithDependencies } from '@prefecthq/vue-compositions'
+  import { computed } from 'vue'
+  import { DeploymentList, DocumentationButton, ExtraInfoModal, ListItemMetaFlowRun, StateListItem } from '@/components'
+  import { useNextFlowRun, useLastFlowRun, useWorkspaceApi, useWorkspaceRoutes, useComponent } from '@/compositions'
+  import { localization } from '@/localization'
+  import { DeploymentsFilter, Flow, FlowsFilter } from '@/models'
+
+  const props = defineProps<{
+    flow: Flow,
+    filter?: FlowsFilter,
+    disabled?: boolean,
+  }>()
+
+  const emit = defineEmits<{
+    (event: 'delete', value: string): void,
+  }>()
+
+  const { FlowMenu } = useComponent()
+  const { attrs, classes, styles } = useAttrsStylesAndClasses()
+  const api = useWorkspaceApi()
+  const routes = useWorkspaceRoutes()
+
+  const { value: expanded } = useLocalStorage(`flow-list-item-${props.flow.id}--expanded`, false)
+
+  const filter = computed(() => {
+    return {
+      ...props.filter,
+      flows: {
+        ...props.filter?.flows,
+        id: [props.flow.id],
+      },
+    }
+  })
+
+  const deploymentsCountSubscriptionArgs = computed<[DeploymentsFilter]>(() => [filter.value])
+  const deploymentsCountSubscription = useSubscriptionWithDependencies(
+    api.deployments.getDeploymentsCount,
+    deploymentsCountSubscriptionArgs,
+  )
+  const deploymentsCount = computed(() => deploymentsCountSubscription.response ?? 0)
```

Comment:
a `useDeploymentsCount` composition would be nice to reduce the boilerplate here and in other places in the app.